services/src/handlers/datasets.rs

```diff
@@ -392,46 +400,71 @@ struct Columns {
     decimal: Vec<String>,
     numeric: Vec<String>,
     textual: Vec<String>,
+    date: Vec<String>,
+}
+
+enum ColumnDataType {
+    Decimal,
+    Number,
+    Text,
+    Date,
+    Unknown,
+}
+
+impl TryFrom<ColumnDataType> for FeatureDataType {
+    type Error = error::Error;
+
+    fn try_from(value: ColumnDataType) -> Result<Self, Self::Error> {
+        match value {
+            ColumnDataType::Decimal => Ok(FeatureDataType::Decimal),
+            ColumnDataType::Number => Ok(FeatureDataType::Number),
+            ColumnDataType::Text => Ok(FeatureDataType::Text),
+            _ => Err(error::Error::NoFeatureDataTypeForColumnDataType),
+        }
+    }
 }
 
-fn detect_columns(layer: &Layer) -> HashMap<String, FeatureDataType> {
+fn detect_columns(layer: &Layer) -> HashMap<String, ColumnDataType> {
     let mut columns = HashMap::default();
 
     for field in layer.defn().fields() {
         let field_type = field.field_type();
-        if let Ok(data_type) = FeatureDataType::try_from_ogr_field_type_code(field_type) {
-            columns.insert(field.name(), data_type);
-        }
+
+        let data_type = match field_type {
+            OGRFieldType::OFTInteger | OGRFieldType::OFTInteger64 => ColumnDataType::Decimal,
+            OGRFieldType::OFTReal => ColumnDataType::Number,
+            OGRFieldType::OFTString => ColumnDataType::Text,
+            OGRFieldType::OFTDate | OGRFieldType::OFTDateTime => ColumnDataType::Date,
+            _ => ColumnDataType::Unknown,
+        };
+
+        columns.insert(field.name(), data_type);
     }
 
     columns
 }
 
-fn column_map_to_column_vecs(columns: &HashMap<String, FeatureDataType>) -> Columns {
+fn column_map_to_column_vecs(columns: &HashMap<String, ColumnDataType>) -> Columns {
     let mut decimal = Vec::new();
     let mut numeric = Vec::new();
     let mut textual = Vec::new();
+    let mut date = Vec::new();
 
     for (k, v) in columns {
         match v {
-            FeatureDataType::Categorical => { // TODO
-            }
-            FeatureDataType::Decimal => {
-                decimal.push(k.clone());
-            }
-            FeatureDataType::Number => {
-                numeric.push(k.clone());
-            }
-            FeatureDataType::Text => {
-                textual.push(k.clone());
-            }
+            ColumnDataType::Decimal => decimal.push(k.clone()),
+            ColumnDataType::Number => numeric.push(k.clone()),
+            ColumnDataType::Text => textual.push(k.clone()),
+            ColumnDataType::Date => date.push(k.clone()),
+            _ => {}
```

Comment:
I guess the only one left is Unknown. If you leave it here, we can extend it and see, that we have to do something here. The wildcard would match everything else.

Reply:
c5ffd6ec608bf584f64d7f98f94be3c48e545821